fw/http.c

```diff
@@ -7575,7 +7575,8 @@ tfw_http_hm_srv_send(TfwServer *srv, char *data, unsigned long len)
 }
 
 /**
- * Calculate the key of an HTTP request by hashing URI and Host header values.
+ * Calculate the key of an HTTP request by hashing URI and vhost name.
+ * If vhost is not yet assigned, fall back to using the Host header.
  */
 unsigned long
 tfw_http_req_key_calc(TfwHttpReq *req)
```

Comment:
You changed only single function, however host header used in other places where key is used. `tfw_http_req_key_calc()` returns key that will be used as key in TDB, but also we do store/compare string key to avoid collisions. See `TFW_CACHE_REQ_KEYITER()`.